tests/boot/grub_test_snapshot.pm

```diff
@@ -23,13 +23,10 @@ sub run() {
     if (get_var("BOOT_TO_SNAPSHOT")) {
         send_key_until_needlematch("boot-menu-snapshot", 'down', 10, 5);
         send_key 'ret';
-        assert_screen("boot-menu-snapshot-list");
-        # in upgrade/migration scenario, we want to boot from snapshot 1 before migration.
-        if (get_var("UPGRADE")) {
-            send_key 'down';
-            save_screenshot;
-        }
-        send_key 'ret';
+        # assuming that the last snapshot is the downgraded version. sle-12 e.g.
+        for (1 .. 35) { send_key "down" }
+        assert_screen("boot-menu-snapshot-list");    # safer side. we cannot use until_sendkey for this.
```

Comment:
thx @dzedro  do we have migration to sles-11 sp4 - to sles-12-sp2 ?

but the problem is the needle matching it-self: needle could match before your solution, because the list could be detected but the cursor is not there.